### ISSUE TITLE
cylc.network.connection_validator: UnboundLocalError: local variable 'user'

### ISSUE THREAD

**Issue by matthewrmshin** [cylc/cylc#1635]
A user is getting a lot of these in `log/suite/err`:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.6/threading.py", line 532, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.6/threading.py", line 484, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/python2.6/site-packages/Pyro/protocol.py", line 1126, in connectionHandler
    if self.getAdapter().handleConnection(conn, self):
  File "/usr/lib/python2.6/site-packages/Pyro/protocol.py", line 840, in handleConnection
    (ok,reasonCode) = tcpserver.newConnValidator.acceptIdentification(tcpserver,conn,token,challenge)
  File "/opt/cylc-6.7.0/lib/cylc/network/connection_validator.py", line 97, in acceptIdentification
    connection.user = user
UnboundLocalError: local variable 'user' referenced before assignment
```

Looks like `ConnValidator.acceptIdentification` is using `user` from an import and as a local variable.

**Comment by benfitzpatrick** [comment]
It is an artefact of my recent change (a42123b) to that file - it needs the line `user = host = ....` back in.

I corrected it as part of the hash change we're trying to make, but it can be done separately.

**Comment by matthewrmshin** [comment]
I am trying to fix it with this change, which makes it easier to see that `USER` and `HOST` are constants and should not be modified locally. Do you want to incorporate this in your hash change?

```diff
diff --git a/lib/cylc/network/connection_validator.py b/lib/cylc/network/connect
index 7a7faf1..4c74bf3 100644
--- a/lib/cylc/network/connection_validator.py
+++ b/lib/cylc/network/connection_validator.py
@@ -38,7 +38,7 @@ except ImportError:
 from cylc.network import NO_PASSPHRASE, PRIVILEGE_LEVELS
 from cylc.config import SuiteConfig
 from cylc.suite_host import is_remote_host
-from cylc.owner import user, host
+from cylc.owner import user as USER, host as HOST
 
 
 # Access for users without the suite passphrase: encrypting the "no passphrase"
@@ -62,6 +62,7 @@ class ConnValidator(DefaultConnValidator):
         logger = logging.getLogger('main')
         is_old_client = False
         # Processes the token returned by createAuthToken.
+        user, host = (USER, HOST)
         try:
             user, host, uuid, prog_name, proc_passwd = token.split(':', 4)
         except ValueError as exc:
@@ -123,4 +124,4 @@ class ConnValidator(DefaultConnValidator):
         prog_name = os.path.basename(sys.argv[0])
         if passphrase is None:
             passphrase = NO_PASSPHRASE
-        return (user, host, str(uuid), prog_name, md5(passphrase).digest())
+        return (USER, HOST, str(uuid), prog_name, md5(passphrase).digest())
```

**Comment by hjoliver** [comment]
Fixed by #1658.
